src/expr.rs

```diff
@@ -416,7 +416,23 @@ pub(crate) fn format_expr(
                 attrs.last().map_or(expr.span.lo(), |attr| attr.span.hi()),
                 expr.span.lo(),
             );
-            combine_strs_with_missing_comments(context, &attrs_str, &expr_str, span, shape, false)
+
+            let allow_extend = if attrs.len() == 1 {
+                let line_len = attrs_str.len() + 1 + expr_str.len();
+                !attrs.first().unwrap().is_doc_comment()
+                    && context.config.inline_attribute_width() >= line_len
+            } else {
+                false
+            };
```

Comment:
This logic is duplicated in a few spots. I think there's an opportunity to encapsulate this logic for reuse.

```rust
/// Determine if we could place a single attribute on the same line as the rewritten AST node.
fn extend_inline_attr(
    attrs: &[ast::Attribute],
    attrs_str: &str,
    rewrite: &str,
    context: &RewriteContext<'_>,
) -> bool {
    if attrs.len() > 1 {
        return false;
    }

    attrs.first().is_some_and(|attr| {
        // +1 = " "
        let line_len = attrs_str.len() + 1 + rewrite.len();
        !attr.is_doc_comment() && context.config.inline_attribute_width() >= line_len
    })
}
```

```suggestion
            let allow_extend = extend_inline_attr(&expr.attrs, &attrs_str, &expr_str, context);
```